### DISCUSSION TITLE
git@github.com: Permission denied (publickey)

### DISCUSSION THREAD

**Discussion by asudyn** [deployphp/deployer#3630] [Help needed]
I'm following [this tutorial](https://www.mitrais.com/news-updates/how-to-create-ci-cd-with-github-action-and-laravel/) to set up a GitHub Action that would trigger deployment of my PHP Laravel app to a server. I've configured the keys on my server correctly and it does connect to it. But it can't clone my github repository.

In deploy PHP I'm corectly setting the repo:
`set('repository', 'git@github.com:asudyn/my-repo.git');`

But when action runs it fails at this part:

```
[prod] git@github.com: Permission denied (publickey).
[prod] fatal: Could not read from remote repository.
```

My repo is private. So maybe I need to add a private key for GitHub somewhere? But I can't find where to do this in docs or online.

My workflow yml and deploy.php are basically the same as in the tutorial with some minor edits to account for latest version changes.

**Comment by antonmedv** [comment]
> So maybe I need to add a private key for GitHub somewhere?

It is called deploy keys. Add it to your server.

**Reply by asudyn** [reply]
As far as I understand, what happens here is happening on a container created by github when the action gets triggered. So I'll have to add these keys to this container, right?

**Reply by antonmedv** [reply]
On the server.
